solr/solrj/src/test/org/apache/solr/client/solrj/TestBatchUpdate.java

```diff
@@ -46,24 +46,28 @@ public static void beforeTest() throws Exception {
 
   @Test
   public void testWithXml() throws Exception {
-    HttpSolrClient client = (HttpSolrClient) getSolrClient();
-    client.setRequestWriter(new RequestWriter());
+    client =
```

Comment:
`client` must be closed in `SolrJettyTestBase`? would be clearer to do `try-with-resources` for each of these?

Reply:
there is a 
```
  @After
  public synchronized void afterClass() throws Exception {
    if (client != null) client.close();
    client = null;
  }
```

There doesn't appear to be any one pattern...

Reply:
I'm of the opinion - we should close the Solr client where we create the Solr client. Since this change is making new Solr clients for each test - they should be `try-with-resources`.